osv2/search.go

```diff
@@ -0,0 +1,251 @@
+package osv2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/opensearch-project/opensearch-go/v2"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+
+	"github.com/CrowdStrike/opensearchtools"
+)
+
+// SearchRequest is a marshable form of [opensearchtools.SearchRequest] specific to the [opensearchapi.SearchRequest] in OpenSearch V2.
```

Comment:
Should be `marshalable` right?  Maybe "serializable" would read better?

Reply:
I prefer serializable, too, except for the fact that Go doesn't use serialize / deserialize but rather marshal/unmarshal.